### ISSUE TITLE
GLFW 3.3 have trouble with multiples keys 

### ISSUE THREAD

**Issue by Hirosam1** [glfw/glfw#1616]
I notice a change when transitioning from GLFW 3.0 to 3.3. GLFW 3.3 doesn't register all keys when they are pressed almost instantly. Before (vs 3.0) all keys were registered by `glfwGetKey()`, with no problems. Now (vs 3.3) when I use `glfwGetKey()`, only the first key pressed is registered and only after the delay of the [repeat keys](https://askubuntu.com/questions/846030/how-to-set-keyboard-repeat-delay-and-speed-in-ubuntu-gnome-16-10) is passed, that is, when the key starts repeating, GLFW register the key. When you add more than 2 keys at the same time, the problem becomes really apparent.

I was wondering, was something I am missing from the versions changes?  Or is this bug?

**I am using Linux (popOS).**

[Here is the code I am working on ](https://github.com/Hirosam1/OpenGL_Basics). in case you want to test it yourself.

**Comment by Hirosam1** [comment]
I tried with the [3.1 version of GLFW](https://github.com/glfw/glfw/releases/tag/3.1), and it seems to be working fine.

**Comment by MyLegGuy** [comment]
i have the same issue.
this is the test code i'm using. it prints key events from glfwSetKeyCallback along with the time.
https://pastebin.com/raw/773hd0tX

on my system, i have `xset r rate 300 30`. initial repeat delay 300, sequential repeats 30.
here is the output if i push and hold the down and left arrow keys at the **same time**:
[1578180561982] 263 is 1 (initial left key press at correct time)
[1578180562282] 264 is 1 (down key press 300 ms later)
[1578180562315] 264 is 2 (down key repeat 30 ms later)
[1578180562347] 264 is 2  
[1578180562381] 264 is 2  
[1578180562406] 263 is 0  
[1578180562406] 264 is 0  
there are no repeats for left arrow key even though it's also held.
when using this particular combination, i can't seem to get the down arrow key to be the correct one and the left arrow key being the repeating one.

if I try 'a' (65) and 'b' (66), then 'a' comes first.
if I try 'b' (66) and 'c' (67), then 'b' comes first.
if I try 'p' (80) and KEY_LEFT (263), then 'p' comes first.
if I try 'q' (81) and 'p' (80), then 'p' comes first.

if I push LEFT (263), Z (90), RIGHT (262), and DOWN (264) at the same time, here's the output:
[1578183452213] 262 is 1
[1578183452513] 264 is 1
[1578183452546] 264 is 2
[1578183452580] 264 is 2
[1578183452614] 264 is 2
[1578183452647] 264 is 2
[1578183452680] 264 is 2
[1578183452713] 264 is 2
[1578183452741] 262 is 0
[1578183452741] 264 is 0
i confirmed using `xev` program that those 4 keys at once work on my keyboard.

this is on arch linux with X11 and the latest release software. except i compiled glfw today.
glxinfo shows `OpenGL version string: 4.6.0 NVIDIA 440.44`
reverting to 3.3 does not fix the problem.
reverting to 3.2.1 does fix the problem.

Here is the output with LEFT, Z, RIGHT, and DOWN at once test with version 3.2.1:
[1578185069503] 262 is 1
[1578185069503] 90 is 1
[1578185069503] 263 is 1
[1578185069503] 264 is 1
[1578185069804] 264 is 2
[1578185069837] 264 is 2
[1578185069870] 264 is 2
[1578185069887] 262 is 0
[1578185069887] 263 is 0
[1578185069887] 264 is 0
[1578185069903] 90 is 0
even in this version, the repeat events only fire for one key. the initial pressed events work though. it appears to be the same key that repeats in my tests with the latest version.

on another updated machine running arch linux with native glfw wayland, this bug does not happen. From glxinfo, that machine has `OpenGL version string: 2.1 Mesa 19.3.1` however, the key callback function never runs multiple times on the same millisecond (possibly due to slower hardware, possibly due to wayland) and the key repeat only works for one key still.

**Comment by hkva** [comment]
After some quick testing, this bug seems to be caused by the following check: https://github.com/glfw/glfw/blob/7486e12f88b2a9039b805356bcbe259c0fca5c2d/src/x11_window.c#L1272

Running the `events` test program and pressing the `U`+`I` keys at the exact same time gives the following output:
```
000000d8 to 1 at 2.801: Key 0x0055 Scancode 0x001e (U) (u) (with no mods) was pressed
000000d9 to 1 at 2.801: Character 0x00000075 (u) input
000000da to 1 at 2.801: Character 0x00000069 (i) input
000000db to 1 at 3.064: Key 0x0055 Scancode 0x001e (U) (u) (with no mods) was released
```
and after completely disabling the check:
```
00000171 to 1 at 4.712: Key 0x0055 Scancode 0x001e (U) (u) (with no mods) was pressed
00000172 to 1 at 4.712: Character 0x00000075 (u) input
00000173 to 1 at 4.712: Key 0x0049 Scancode 0x001f (I) (i) (with no mods) was pressed
00000174 to 1 at 4.712: Character 0x00000069 (i) input
00000175 to 1 at 4.904: Key 0x0055 Scancode 0x001e (U) (u) (with no mods) was released
00000176 to 1 at 4.904: Key 0x0049 Scancode 0x001f (I) (i) (with no mods) was released
```

**Comment by elmindreda** [comment]
Thank you for reporting this. It should be fixed now with 9a3664b4a992a5a9b669179098542b0f882dece7.
